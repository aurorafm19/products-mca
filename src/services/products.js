const apiUrl = 'https://front-test-api.herokuapp.com/api/';
const headers = { 'Content-Type': 'application/json'};

export function getProductlist() {
    return fetch(`${apiUrl}product`, {method: 'GET', headers});
}

export function getProductDetail(id) {
    return fetch(`${apiUrl}product/${id}`, {method: 'GET', headers})
    .then((response) => response.json());
}

export function addProduct(product) {
    fetch(`${apiUrl}/cart`, {method: 'POST', headers, body: JSON.stringify(product)} )
}

export function filterBySearchValue(searchValue, productList) {
    let productsFiltered = [];
    if (searchValue.length > 0) {
        productsFiltered = productList.filter(product =>product.brand === searchValue || product.model === searchValue )
    }

    return productsFiltered;
}
