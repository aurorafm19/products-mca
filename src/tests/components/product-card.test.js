import { render} from '@testing-library/react';
import ProductCard from '../../components/product-card';
import {PRODUCTS} from '../mocks/products';

test('renders the component', () => {
    const {container} = render(<ProductCard product={PRODUCTS[0]} />);

    expect(container).toMatchSnapshot();
});

test('render the component when price and model/brand dont exist', () => {
    const {container} = render(<ProductCard product={PRODUCTS[1]} />);

    expect(container).toMatchSnapshot();
})