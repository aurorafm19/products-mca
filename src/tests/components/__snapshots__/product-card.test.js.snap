// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the component when price and model/brand dont exist 1`] = `
<div>
  <a
    class="product-card card"
    href="/product/ke-gsQbO8qH9PQ-zcdiAJ"
  >
    <img
      alt="mobile"
      class="product-card-image"
    />
    <div
      class="product-card-content"
    >
      <div
        class="product-card-not-info"
      >
        not info available
      </div>
      <div
        class="product-card-not-info"
      >
        not info available
      </div>
    </div>
  </a>
</div>
`;

exports[`renders the component 1`] = `
<div>
  <a
    class="product-card card"
    href="/product/ke-gsQbO8qH9PQ-zcdiAJ"
  >
    <img
      alt="mobile"
      class="product-card-image"
    />
    <div
      class="product-card-content"
    >
      <div
        class="product-card-title"
      >
        Liquid Zest plus
        -
        Acer
      </div>
      <div
        class="product-card-desc"
      >
        200
        $
      </div>
    </div>
  </a>
</div>
`;
